src/Repository/FileRepository.hpp

```diff
@@ -33,16 +34,25 @@ Copyright_License {
  * The announced contents of a remote file repository.  These are the
  * files that can be downloaded by the XCSoar user.
  */
-struct FileRepository {
+class FileRepository {
```

Comment:
Your commit message says
> FileRepository is made a class because it exerts behaviour instead of only holding data

... but I don't understand this deduction.
Having behaviour is not what makes the difference between `struct` and `class`. A `struct` can have behavior as well - and in fact, the only difference between the two is the default member access.

Reply:
Yes it's not worded in the best way possible.
I was taught that in C++ structs should be primarily used for holding raw data and as soon as an abstraction of the underlying data or behavior is added (where private members are used) a class is more appropriate.
But I can understand that there are different schools of thought so I'm also good with a struct.

Reply:
It's perfectly fine to convert struct to class if you want to wrap its implementation details in nicer public methods. But that was not your declared reason to do it, and it was hidden deep inside a complex patch, so I couldn't deduce your "real" reason from your code.